tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/cassandra/CassandraTableManager.java

```diff
@@ -0,0 +1,227 @@
+import com.teradata.tempto.fulfillment.table.TableHandle;
+import com.teradata.tempto.fulfillment.table.TableInstance;
+import com.teradata.tempto.fulfillment.table.TableManager;
+import com.teradata.tempto.fulfillment.table.jdbc.RelationalDataSource;
+import com.teradata.tempto.fulfillment.table.jdbc.RelationalTableDefinition;
+import com.teradata.tempto.internal.fulfillment.table.TableName;
+import com.teradata.tempto.internal.fulfillment.table.TableNameGenerator;
+import com.teradata.tempto.internal.query.CassandraQueryExecutor;
+import org.slf4j.Logger;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.teradata.tempto.fulfillment.table.MutableTableRequirement.State.CREATED;
+import static com.teradata.tempto.fulfillment.table.MutableTableRequirement.State.LOADED;
+import static com.teradata.tempto.fulfillment.table.MutableTableRequirement.State.PREPARED;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@TableManager.Descriptor(tableDefinitionClass = RelationalTableDefinition.class, type = "CASSANDRA")
+@Singleton
+public class CassandraTableManager
+        implements TableManager<RelationalTableDefinition>
+{
+    private static final Logger LOGGER = getLogger(CassandraTableManager.class);
+
+    private static final int BATCH_SIZE = 10_000;
+
+    private final TableNameGenerator tableNameGenerator;
+    private final CassandraQueryExecutor queryExecutor;
+    private final String databaseName;
+    private final String defaultKeySpace;
+
+    @Inject
+    public CassandraTableManager(TableNameGenerator tableNameGenerator,
+            @Named("databaseName") String databaseName,
+            Configuration configuration)
+    {
+        this.tableNameGenerator = requireNonNull(tableNameGenerator, "tableNameGenerator is null");
+        this.queryExecutor = new CassandraQueryExecutor(configuration);
+        this.databaseName = requireNonNull(databaseName, "databaseName is null");
+        this.defaultKeySpace = configuration.getStringMandatory("databases.cassandra.default_schema");
+    }
+
+    @Override
+    public TableInstance<RelationalTableDefinition> createImmutable(RelationalTableDefinition tableDefinition, TableHandle tableHandle)
+    {
+        TableName tableName = createImmutableTableName(tableHandle);
+        checkArgument(tableName.getSchema().isPresent(), "Cassandra does not like you when you don't have keyspace defined");
+
+        if (!queryExecutor.tableExists(tableName.getSchema().get(), tableName.getInternalNameInDatabase())) {
+            executeQueryIgnoreError(tableDefinition.getCreateTableDDL(tableName.getNameInDatabase()));
```

Comment:
We don't support all data types in Cassandra.
`CassandraQueryExecutor.getJDBCType` throws an exception when there's no mapping for data type. We don't care about output of `executeQuery(SQL)` in DDL because we want to be able to create a table even with types we don't support.
Though `CassandraQueryExecutor` should throw that exception in case one wanted to read that output.